src/impl_/pymethods.rs

```diff
@@ -175,12 +182,19 @@ impl PyMethodDef {
             },
         };
 
-        Ok(ffi::PyMethodDef {
-            ml_name: get_name(self.ml_name)?.as_ptr(),
+        let name = get_name(self.ml_name)?;
+        let doc = get_doc(self.ml_doc)?;
+        let def = ffi::PyMethodDef {
+            ml_name: name.as_ptr(),
             ml_meth: meth,
             ml_flags: self.ml_flags,
-            ml_doc: get_doc(self.ml_doc)?.as_ptr(),
-        })
+            ml_doc: doc.as_ptr(),
+        };
+        let destructor = PyMethodDefDestructor {
+            name: Some(name),
```

Comment:
Is `PyMethodDefDestructor` ever constructed with `None` values? Can the outer `Option` layer be dropped entirely?

Reply:
Will drop the option layer.